sbi/inference/snpe/snpe_base.py

```diff
@@ -252,6 +252,7 @@ def train(
             self._neural_net = self._build_neural_net(
                 theta[self.train_indices], x[self.train_indices]
             )
+            self._neural_net.to(self._device)
```

Comment:
will fail when data is not on the `device` yet. 
not needed in general when data is assumed to be moved to `device` only during training.

Reply:
in the best of all worlds, `self._build_neural_net` would be capable of ingesting `self._device` to place the NN on the device. This would be a much more clean solution. But neither `utils.posterior_nn` can do that nor a user provided function.